File: 1512-number-of-good-pairs/goodpairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        pairs = {}
        count = 0
        
        for num in nums:
            if num in pairs:
                count+=pairs[num]
                pairs[num]+=1
            else:
                pairs[num]=1
        
        
        return count

File: 1512-number-of-good-pairs/goodpairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
        let numbers = {}
        let count = 0
        
        for (let num of nums){
            if(numbers[num]) {
                count += numbers[num]
                numbers[num]++
            } else {
                numbers[num] = 1 }
        }
    
        return count
};